presto-verifier/src/main/java/com/facebook/presto/verifier/framework/ExplainVerification.java

```diff
@@ -88,10 +95,27 @@ protected ExplainMatchResult verify(
             ChecksumQueryContext controlChecksumQueryContext,
             ChecksumQueryContext testChecksumQueryContext)
     {
-        checkArgument(controlQueryResult.isPresent(), "control query plan is missing");
         checkArgument(testQueryResult.isPresent(), "test query plan is missing");
 
-        JsonRenderedNode controlPlan = planCodec.fromJson(getOnlyElement(controlQueryResult.get().getResults()));
+        JsonRenderedNode controlPlan;
+        if (!skipControl || saveSnapshot) {
+            checkArgument(controlQueryResult.isPresent(), "control query plan is missing");
+            String result = getOnlyElement(controlQueryResult.get().getResults());
+            if (saveSnapshot) {
+                snapshotQueryConsumer.accept(new SnapshotQuery(getSourceQuery().getSuite(), getSourceQuery().getName(), isExplain, result));
+                return new ExplainMatchResult(MATCH);
```

Comment:
nit: rest of the match results have some more constructor args to identify that the match result was a noop.
Maybe consider adding a MatchType.NOOP to handle the saving snapshot logic cleanly instead of returning a MATCH